Dockerfile

```diff
@@ -1,5 +1,5 @@
 ARG BUILDPLATFORM="linux/amd64"
-ARG BUILDERIMAGE="golang:1.17"
+ARG BUILDERIMAGE="golang:1.20"
```

Comment:
Let's use `cgr.dev/chainguard/go`, wdyt?

Reply:
I think that's a great idea, but I think this should be tackle in another PR, as I think it's need to be discussed separately. For example, do you know why [`rekor`](https://github.com/sigstore/rekor/blob/main/Dockerfile#L16) or [`fulcio`](https://github.com/sigstore/fulcio/blob/main/Dockerfile#L16) are using `golang:1.20` and not the `cgr.dev/chainguard/go` one you mention?

Reply:
Good catch! I have no idea, but maybe we should update those images too. Also I don't see any particular reason to not switch the image to chainguard's we can tackle in this PR.

Reply:
I think the question is "Is there any particular reason not doing that in another PR?" and "do we really want this blocking this current PR?".

This current PR is doing `go:1.17` --> `go:1.20`. Doing `go` --> `` is adding other tests, commit, dicussion and validation, I filed this in another issue there: https://github.com/sigstore/cosign-gatekeeper-provider/issues/27.

I agree with you "which should merge this current PR ASAP to unblock #26 and the use of this project in its current state.

Hope that makes sense and we could have an approval on this current PR as-is.

Reply:
Thanks for the issue! Let's discuss this out there to unblock your PR. Strong +1 to your last statement.